### PR TITLE
Refactors Zips service to medianIncome service.

### PR DESCRIPTION
**Pull request by xoddong**
-Changes file/service name to medianIncome to avoid confusion with the zipsOfCity service.
-On the client side:
1. passes in zipcode and rangeInc params during GET.
2. removes the income filter (moves to the server side which is the reason for point number 1).
-On the server side:
1. Changes fileRead from async to sync to remain DRY after adding filter.
2. Moves filter to server side because we can minimize the iteration from two to one by having it in one place.
3. Returns an array of objects(zipcode && median income as properties) that fit the income range specified in the search.

### CODE REVIEW

**Comment by xoddong** [comment]
To test:

- In home.client.controller, use the service 'medianIncome' and its filterByZipcodes function. Pass in a fake array of zipcodes and an array of median income range.
-Include a call back to console.log the return data.
-For example, medianIncome.filterByZipcodes([94110, 94118], [40000, 52000], function(data){
  console.log(data);
}).

**Comment by tesk9** [comment]
The changes (to titles and function names and such) are resulting in failing tests when running grunt test. I haven't tested it by hand yet, but I don't think we should merge code that fails any of our tests, you know?

File: public/modules/core/services/medianIncome.client.service.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+          }).success(function(medIncArr) {
+            // medInc is an object with two properties (zipcode and medianIncome)
+            cb(medInc);
+          }).error(function() {
+            return;
+          });
+      }
+    };
+  }
+
```
Comment (tesk9) [inline public/modules/core/services/medianIncome.client.service.js]: Love the comments. Such readable code!

File: public/modules/core/services/medianIncome.client.service.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+// Zip code service for filtering results
+angular.module('core').service('medianIncome', [ '$http',
+  function($http) {
+    return {
+      // zipcodes paramter of filteredZipCodes is an array of zipcode(s).
```
Comment (tesk9) [inline public/modules/core/services/medianIncome.client.service.js]: "parameter"

**Comment by xoddong** [comment]
ook i'll check it out to see what is causing the problem!

**Pull request merged**